### PR TITLE
HDDS-7245. Recon: Add button to trigger OM DB Sync (Frontend)

### PR DESCRIPTION
**Pull request by smitajoshi12**
What changes were proposed in this pull request?
After Clicking Button IT should trigger om Sync and disable button show message in tool tip.

What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7245

## How was this patch tested?
Manually

image

### CODE REVIEW

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx [hunk 1]
```diff
@@ -49,9 +51,12 @@ class AutoReloadPanel extends React.Component<IAutoReloadPanelProps> {
         >
           {moment(lastRefreshed).format('LT')}
         </Tooltip>
-      );
-
-      const omDBDeltaFullToolTip = <span>
+       );
+    
+     const omSyncStatusDisplay = omStatus === '' ? '' : omStatus ? <div>OM DB update is successfully triggered.</div> : <div>OM DB update is already running.</div>;
```
Comment (DaveTeng0) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:56]: does the second _omStatus_ variable check if the string is true or false?

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:56]: omStatus may contain three values  Empty space when we are not clicking to button and true and false depending upon  END Point Response.

sync trigger endpoint returns false: update button tooltip can show "OM DB update is already running", and becomes gray and unclickable
sync trigger endpoint returns true: update button tooltip can show "OM DB update is successfully triggered", also becomes gray and unclickable

**Comment by myskov** [comment]
Users of the Recon are not developers, and they might don't know what `OM DB` is. Maybe something like "Cluster sync"?

**Review by myskov — COMMENTED** [review]
Users of the Recon are not developers, and they might don't know what OM DB is. Maybe something like "Cluster sync"?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx (hunk 1 above)
Comment (DaveTeng0) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:56]: ohh i see, thanks!!

**Comment by dombizita** [comment]
> Users of the Recon are not developers, and they might don't know what OM DB is. Maybe something like "Cluster sync"?

Hi @myskov, thanks for the review. Recon also maintains a DB for SCM which later can be added here as well. This is why we decided to differentiate these two syncs and name it OM DB sync (also we only trigger a sync between OM and Recon with the button, so other cluster related informations won't get refreshed by this). I think "Cluster sync" could be misleading, but I am open to other names. :)

**Comment by smengcl** [comment]
@smitajoshi12 Would you resolve the merge conflict in `autoReloadPanel.tsx`?

**Comment by smengcl** [comment]
Thanks @smitajoshi12 for the patch, @DaveTeng0 for the review.

@dombizita and @myskov we can continue the discussion in the follow-up jira HDDS-7505.

**Pull request merged**